.gitlab-ci.yml

```diff
@@ -0,0 +1,107 @@
+stages:
+  - build
+  - package
+  - publish
+
+image:                  parity/rust:gitlab-ci
+
+cache:
+  key:                  "${CI_JOB_NAME}"
+  paths:
+    - ${CI_PROJECT_DIR}/target/
+    - ${CI_PROJECT_DIR}/cargo/
+    - node_modules/
+    - .cache
+    - $HOME/.cache
+
+.releaseable_branches:  # list of git refs for building GitLab artifacts (think "pre-release binaries")
+  only:                 &releaseable_branches
+    - schedules # Our nightly builds from schedule, on `master`
+    - "v*" # Our version tags
+    - ci-package
+    - next
+    - tags
+
+.publishable_branches:  # list of git refs for publishing builds to the "production" locations
+  only:                 &publishable_branches
+    - schedules # Our nightly builds from schedule, on `master`
+    - "v*" # Our version tags
+    - ci-package
+
+before_script:
+  - npm install --global yarn --cache-min 999999999
+  - npm install
+  - export VERSION=$(node -p "require('./package.json').version")
+
+# Stage - build
+build:
+  stage:                build
+  script:
+    - scripts/lint.sh
+    # - scripts/test.sh
+    - scripts/build.sh
+    - if [ "$CI_BUILD_REF_NAME" == "master" ]; then scripts/push-precompiled.sh; fi
```

Comment:
I put the same line in `push-precompiled.sh` script, here is maybe not necessary? Or is it cleaner to put it here?

However I used `$CI_COMMIT_REF_NAME `, don't know the difference between CI_COMMIT_REF_NAME and CI_BUILD_REF_NAME.